read_motif_v6.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Nov  2 01:28:58 2020
+
+@author: YB Moon (ymoon06)
+"""
+"""
+input:
+    motif = motif sequence. must be string and big character
+    read= = read sequence. must be string and big character
+    assume motif < read
+    min_overlap = min threshold for match
+    full_contain= boolean parameter. If True, only report matches that are
+                  fully contained in the read
+"""
+"""
+output:
+    returns a list of tuple(a,b) where:
+    a is overlap start position of read
+    b is fraction of motif that overlaps
+"""
+
+
+def find_overlaps(motif, read, min_overlap, full_contain=False):
+    """check the type of arguments """
+    if not isinstance(motif, str):
+        raise TypeError('Incorrect argument type: motif')
+    if not isinstance(read, str):
+        raise TypeError('Incorrect argument type: read')
+    if not isinstance(min_overlap, int):
+        raise TypeError('Incorrect argument type: min_overlap')
+    if not isinstance(full_contain, bool):
+        raise TypeError('Incorrect argument type: full_contain')
+    """check value of arguments"""
+    if len(motif) == 0:
+        raise ValueError('length of motif must be bigger than 0')
+    if len(read) == 0:
+        raise ValueError('length of read must be bigger than 0')
+    if min_overlap < 1:
+        raise ValueError('min_overlap must be longer than 1')
+    if len(read) < len(motif):
+        raise ValueError('read should be >= than motif')
+    if not motif.isupper():
+        raise ValueError('motif should be all big characters')
+    if not read.isupper():
+        raise ValueError('read should be all big characters')
+    """compute partial overlaps of the motif at the start of the read"""
```

Comment:
Use `"""` comments only for docstrings, otherwise use `#` comments. Docstrings are the documentation strings that are expected as the first statement of a module, class, function and method.